### PR TITLE
refactor!: Make the col! macro a preferred alias of column_expr!

### PR DESCRIPTION
**Pull request by scovich**
## What changes are proposed in this pull request?

The `column_expr!` and `col!` macros both accept the same kinds of args and produce column name expressions, but with subtly different semantics for some invocation patterns. Given `const FOO: &str = "foo"` and `const BAR_BAZ: &str = "bar.baz"`:

|arg(s)|col! result|column_expr! result|comment
|-|-|-|-|
|`"a.b"`|`["a", "b"]`|`["a", "b"]`|Equivalent
|`"a.b", "c"` | `["a.b", "c"]` | `["a", "b", "c"]` | `col!` only splits string literals on `.` for 1-arg invocation
|`FOO`|`["foo"]`|`["foo"]`|`column_expr!` accepts `FOO` because it is contains simple path segment (no `.` or special chars)
|`BAR_BAZ`|`["bar.baz"]`|compilation error|`column_expr!` rejects `BAR` because it contains `.`
|`foo`|`[foo]`|compilation failure|`col!` performs runtime value interpolation with no visual syntax to request it

Note: `column_expr!` originally supported only a single string literal arg. That limitation was especially painful for constant args and at least partly motivated the introduction of the `col!` macro. However, `column_expr!` now handles multiple arguments and also handles both string literals and constants, which brings it much closer in capabilities to `col!`.

An audit of all `col!` and `column_expr!` call sites shows:
|arg pattern|# col!|# col_expr!|comment
|-|-|-|-|
|single-arg simple string literal, e.g. `"x"`|526|567|
|single-arg dot split string literal, e.g. `"a.b"`|8|32|
|single-arg with constant arg, e.g. `FOO`|10|0|
|single-arg with runtime arg, e.g. `name`|4|-|`column_expr!` forbids this case
|multi-arg with all constant args, e.g. `FOO, BAR`|16|0|
|multi-arg mixing constants and literals, e.g. `FOO, "a"`|1|1|The `col!` case is a doc test showing that `"a.b"` is not split
|multi-arg involving runtime values, e.g. `"a", name`|0|-|`column_expr!` forbids this case

This is good news, because the most ambiguous case (`col!("a.b", "c")` not splitting on dots) does not actually arise outside the doc test that demonstrates the corner case.

Because the behavior of `col!` is generally more surprising (and less verifiable) than `column_expr!`, this PR changes the former into an alias of the latter. That allows everyone to use the shorter `col!` name but with clear semantics:
* 1+ arguments supported
* String literals containing dots are always split
* Constant values are allowed, but only if they do not contain dots or other special characters (enforced at compilation time)
* Runtime values are not allowed, because they cannot be validated. 

A selection of `column_expr!` sites are also changed to `col!`, to demonstrate the advantages of the shorter name. To avoid excessive churn in a behavior change PR, we leave a full sweep of the 600+ remaining `column_expr!` sites to future work.

Future work could add proper runtime interpolation support to the macros, so that e.g. `col!({name})` and `col!({"a.b"})` would clearly indicate that the caller requests runtime interpolation for a single path segment, producing `Expression::column([name])` and `Expression::column(["a.b"])` respectively. This would allow to "reclaim" the four `col!` sites this PR had to rewrite as `Expression::column`. It would also allow to retire the `joined_column_*` family of macros in favor of e.g. `col!("a", {name}, FOO)` (which produces `Expression::column(["a", name, FOO])`.

Future work could also consider removing the literal string-split capability altogether, since `col!("a.b.c")` can already be expressed by `col!("a", "b", "c")`. That said, the former easier to read and _much_ easier to type (`", "` requires pressing and releasing the shift key twice in three keystrokes).

### This PR affects the following public APIs

`col!` macro changes behavior to match `column_expr!`, and runtime values are no longer accepted. Use `Expression::column` instead for those.

The `column_expr!` macro remains usable, but is now `doc(hidden)` to steer new users toward the shorter name.

## How was this change tested?

Compilation and lots of unit tests.

### CODE REVIEW

File: kernel/src/checkpoint/sidecar/mod.rs [hunk 1]
```diff
@@ -218,7 +218,7 @@ impl SidecarSplitter {
                 checkpoint_data_schema
                     .fields()
                     .filter(|f| f.name != ADD_NAME && f.name != REMOVE_NAME)
-                    .map(|f| Predicate::is_not_null(col!(f.name.as_str()))),
+                    .map(|f| Expression::column([&f.name]).is_not_null()),
```
Comment (scovich) [inline kernel/src/checkpoint/sidecar/mod.rs:221]: One of the runtime interpolation patterns that had to be removed

File: kernel/src/expressions/column_names.rs
Comment (scovich) [inline kernel/src/expressions/column_names.rs:1]: The diff here is messy, but all I did was:
* Delete the entire `__col` macro definition, including its docs and re-exports
* Add a missing doc comment to `__column_expr` definition (macro definition unchanged!)
* Re-export `__column_expr` as both `col` and `column_expr` (`doc(hidden)`)

File: kernel/src/scan/transform_spec.rs [hunk 2]
```diff
@@ -144,8 +142,8 @@ pub(crate) fn get_transform_expr(
                     Error::generic("Asked to generate RowIds, but no baseRowId found.")
                 })?;
                 let expr = Arc::new(Expression::coalesce([
-                    col!(field_name),
-                    lit(base_row_id) + col!(row_index_field_name),
+                    Expression::column([field_name]),
+                    lit(base_row_id) + Expression::column([row_index_field_name]),
```
Comment (scovich) [inline kernel/src/scan/transform_spec.rs:146]: More runtime interpolation (and one more below)

**Review by scovich — COMMENTED** [review]
Comments to aid reviewers

**Review by OussamaSaoudi — APPROVED** [review]

File: kernel/src/scan/transform_spec.rs (hunk 2 above)
Comment (thinh2) [reply kernel/src/scan/transform_spec.rs:146]: just curious, why do we need to use the `Expression::column` instead of the `col!` ?

**Review by thinh2 — APPROVED** [review]

File: kernel/src/scan/transform_spec.rs (hunk 2 above)
Comment (scovich) [reply kernel/src/scan/transform_spec.rs:146]: `col!` no longer accepts runtime values with this change
